File: app/src/main/java/fr/corenting/edcompanion/network/retrofit/EDSMRetrofit.java
package fr.corenting.edcompanion.network.retrofit;

import java.util.List;

import fr.corenting.edcompanion.models.apis.EDSM.EDSMCreditsResponse;
import fr.corenting.edcompanion.models.apis.EDSM.EDSMPositionResponse;
import fr.corenting.edcompanion.models.apis.EDSM.EDSMRanksResponse;
import fr.corenting.edcompanion.models.apis.EDSM.EDSMServerStatusResponse;
import fr.corenting.edcompanion.models.apis.EDSM.EDSMSystemResponse;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface EDSMRetrofit {
    @GET("api-logs-v1/get-position")
    Call<EDSMPositionResponse> getPosition(@Query("apiKey") String apiKey,
                                           @Query("commanderName") String commanderName);

    @GET("api-commander-v1/get-credits")
    Call<EDSMCreditsResponse> getCredits(@Query("apiKey") String apiKey,
                                         @Query("commanderName") String commanderName);

    @GET("api-commander-v1/get-ranks")
    Call<EDSMRanksResponse> getRanks(@Query("apiKey") String apiKey,
                                     @Query("commanderName") String commanderName);

    @GET("api-status-v1/elite-server")
    Call<EDSMServerStatusResponse> getServerStatus();

    @GET("api-v1/systems")
    Call<List<EDSMSystemResponse>> getSystems(@Query("systemName") String systemName,
                                              @Query("showId") int showId,
                                              @Query("showInformation") int showInformation,
                                              @Query("showPermit") int showPermit);
}

File: app/src/main/java/fr/corenting/edcompanion/network/retrofit/EDSMRetrofit.kt
package fr.corenting.edcompanion.network.retrofit

import fr.corenting.edcompanion.models.apis.EDSM.*
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Query

interface EDSMRetrofit {

    @GET("api-logs-v1/get-position")
    suspend fun getPosition(
        @Query("apiKey") apiKey: String?,
        @Query("commanderName") commanderName: String?
    ): EDSMPositionResponse

    @GET("api-commander-v1/get-credits")
    suspend fun getCredits(
        @Query("apiKey") apiKey: String?,
        @Query("commanderName") commanderName: String?
    ): EDSMCreditsResponse

    @GET("api-commander-v1/get-ranks")
    suspend fun getRanks(
        @Query("apiKey") apiKey: String?,
        @Query("commanderName") commanderName: String?
    ): EDSMRanksResponse

    @GET("api-status-v1/elite-server")
    suspend fun getServerStatus(): EDSMServerStatusResponse

    @GET("api-v1/systems")
    fun getSystems(
        @Query("systemName") systemName: String?,
        @Query("showId") showId: Int,
        @Query("showInformation") showInformation: Int,
        @Query("showPermit") showPermit: Int
    ): Call<List<EDSMSystemResponse>>
}